<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CSVLineSaver_Widget</class>
 <widget class="QWidget" name="CSVLineSaver_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label_description">
     <property name="text">
      <string>Description: CSV is a human readable spreadsheet format. For Line Data, you can save as a single file with all timestamps, or as multiple files (one per timestamp).</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="save_mode_groupbox">
     <property name="title">
      <string>Save Mode</string>
     </property>
     <layout class="QVBoxLayout" name="save_mode_layout">
      <item>
       <widget class="QRadioButton" name="single_file_radio">
        <property name="text">
         <string>Single File - All timestamps in one CSV file</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="multi_file_radio">
        <property name="text">
         <string>Multi File - One CSV file per timestamp (first line only)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_filename">
       <property name="text">
        <string>Filename/Directory:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="save_filename_edit">
       <property name="text">
        <string>lines_output.csv</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_delimiter_saver">
       <property name="text">
        <string>Delimiter:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="delimiter_saver_combo">
       <item>
        <property name="text">
         <string>Comma</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Space</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Tab</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_line_ending">
       <property name="text">
        <string>Line Ending:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="line_ending_combo">
       <item>
        <property name="text">
         <string>LF (\n)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CRLF (\r\n)</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_precision">
       <property name="text">
        <string>Precision:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_precision">
       <item>
        <widget class="QSpinBox" name="precision_spinbox">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>5</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_precision_example">
         <property name="text">
          <string>e.g., 1.2</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_precision">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QCheckBox" name="save_header_checkbox">
       <property name="text">
        <string>Save Header</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_header_text">
       <property name="text">
        <string>Header Text:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="header_text_edit">
       <property name="text">
        <string>Frame,X,Y</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_frame_padding">
       <property name="text">
        <string>Frame ID Padding:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QSpinBox" name="frame_padding_spinbox">
       <property name="toolTip">
        <string>Number of digits for frame number in filename (e.g., 7 for 0000001.csv). Only used for multi-file mode.</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
       <property name="value">
        <number>7</number>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <widget class="QCheckBox" name="overwrite_existing_checkbox">
       <property name="text">
        <string>Overwrite existing files</string>
       </property>
       <property name="toolTip">
        <string>If checked, existing files will be overwritten. If unchecked, existing files will be skipped. Only applies to multi-file mode.</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPushButton" name="save_action_button">
     <property name="text">
      <string>Save to CSV</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 